cmd/parse-genesis/parseGenesis.go

```diff
@@ -0,0 +1,43 @@
+package parsegenesis
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/forbole/bdjuno/v2/cmd/parse-genesis/utils"
+	"github.com/forbole/juno/v2/cmd/parse"
+	"github.com/spf13/cobra"
+)
+
+// NewParseGenesisCmd returns the Cobra command allowing to parse the genesis file
+func NewParseGenesisCmd(parseCfg *parse.Config) *cobra.Command {
+	return &cobra.Command{
+		Use:     "parse-genesis [optional: module names]",
+		Short:   "Parse genesis file. To parse specific modules, input module names as arguments",
+		Example: "bdjuno parse-genesis auth bank consensus gov history staking",
+		PreRunE: parse.ReadConfig(parseCfg),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			genesisDoc, genesisState, err := utils.GetGenesisDocAndState()
+			if err != nil {
+				return fmt.Errorf("error while getting genesis doc or state: %s", err)
+			}
+
+			registeredModules, err := utils.GetRegisteredModules(parseCfg)
```

Comment:
Instead of defining a new function that mimics the behavior of an already existing function, we should leverage the code that already exists instead. In this case, the `GetREgisteredModules` function can be easily replaced with the following lines: 
```go
parseCtx, err := parse.GetParsingContext(parseCfg)
if err != nil {
    return err
}
```

Looking at the value returned from `GetParsingContext`, we can see that the returned `Context` object already contains a list of enabled modules: 
```go
type Context struct {
    EncodingConfig *params.EncodingConfig
    Node           node.Node
    Database       database.Database
    Logger         logging.Logger
    Modules        []modules.Module
}
```

Once we have the context, we can then simply do: 
```go
for _, module := range modules {
    if genesisModule, ok := module.(modules.GenesisModule) { 
        genesisModule.ParseGenesis(...)
    }
}
```

Reply:
got it. will use parseCtx instead to get the registered modules. 

I will also remove the logic that user can insert module name as args to only parse certain genesis modules. I assume specifying in the config.yaml would be enough.

Reply:
@huichiaotsou I think we can keep the logic of only specifying some modules to be parsed I believe. It's going to be useful if we add new modules on the way. We can however simplify it, I can show you how in the next review after all the other comments are solved